src/main/scala/org/constellation/domain/transaction/TransactionChainService.scala

```diff
@@ -2,16 +2,15 @@ package org.constellation.domain.transaction
 
 import cats.effect.Concurrent
 import cats.implicits._
-import io.chrisdavenport.log4cats.Logger
 import org.constellation.primitives.concurrency.SingleRef
 
-class TransactionChainService[F[_]: Concurrent: Logger] {
+class TransactionChainService[F[_]: Concurrent] {
   // TODO: Make sure to clean-up those properly
   private[domain] val lastTransactionCount: SingleRef[F, Long] = SingleRef(0)
   private[domain] val addressCount: SingleRef[F, Map[String, Long]] = SingleRef(Map.empty[String, Long])
```

Comment:
I only see potential problem that this will hold all addresses in memory and wondering if they could fit ?